src/Bicep.LangServer.IntegrationTests/TelemetryTests.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading.Tasks;
+using Bicep.Core.Syntax;
+using Bicep.Core.Text;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.LanguageServer.Telemetry;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Bicep.LangServer.IntegrationTests
+{
+    [TestClass]
+    [SuppressMessage("Style", "VSTHRD200:Use \"Async\" suffix for async methods", Justification = "Test methods do not need to follow this convention.")]
+    public class TelemetryTests
+    {
+        [NotNull]
+        public TestContext? TestContext { get; set; }
+
+        [TestMethod]
+        public async Task ValidateDeclarationSnippetCompletionItemContainsCommandWithTelemetryInformation()
+        {
+            var syntaxTree = SyntaxTree.Create(new Uri("file:///main.bicep"), string.Empty);
+            var client = await IntegrationTestHelper.StartServerWithTextAsync(string.Empty, syntaxTree.FileUri);
+
+            var completions = await client.RequestCompletion(new CompletionParams
+            {
+                TextDocument = new TextDocumentIdentifier(syntaxTree.FileUri),
+                Position = TextCoordinateConverter.GetPosition(syntaxTree.LineStarts, 0),
+            });
+
+            CompletionItem completionItem = completions.Where(x => x.Kind == CompletionItemKind.Snippet && x.Label == "res-aks-cluster").First();
+            Command? command = completionItem.Command;
+            TelemetryEvent? telemetryEvent = command?.Arguments?.First().ToObject<TelemetryEvent>();
```

Comment:
What happens in the fluent assertions if command is null?  Does it fail or get ignored?

Reply:
Added some null checks to make sure the test fails if something is null:
https://github.com/Azure/bicep/commit/54c6a5bf92fbcc1cf8f964d513e59dd29bf08ae3

Reply:
@anthony-c-martin What's the best way to handle this with fluent assertions?   The "!" not-null assertion doesn't seem optimal either.

Reply:
Switched to use assertions as part of this change:
https://github.com/Azure/bicep/commit/43337953f0cb9fe0d8c4386e00322b231ce9e766

Reply:
Good call.  We tend to use `!.` instead of `?.` in tests for this exact reason. As @StephenWeatherford mentioned, lines 42-44 won't assert if `command` and `telemetryEvent` are null.

With `!.` you'll at least get an unhandled exception rather than silently passing the test.

Reply:
Updated the test here:
https://github.com/Azure/bicep/commit/e4dd5cb53c39092925cf801a49c8ec58751f3666